src/app/shared/auth-guard.service.ts

```diff
@@ -70,4 +70,17 @@ export class AuthService {
     );
   }
 
+  isAuthenticated(): Observable<boolean> {
+    return this.getSession$().pipe(
+      take(1),
+      map(sessionInfo => {
+        const isLoggedIn = !!sessionInfo.userCtx.name;
+        if (!isLoggedIn) {
+          this.router.navigate([ '/login' ], { queryParams: { returnUrl: this.router.url } });

```

Comment:
I think we should throw an error here. A simple "Not authorized" message would probably work. This will break the observable and avoid running the subscription function when we call this. Makes it so we don't need to check if the user is logged in every time this gets called.